source/PlayerInfo.cpp

```diff
@@ -49,6 +49,26 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 
 using namespace std;
 
+namespace {
+	// Move the flagship to the start of your list of ships. It does not make sense
+	// that the flagship would change if you are reunited with a different ship that
+	// was higher up the list.
+	// This function should not be used while other functions have direct references
+	// to the shared_pointer elements in this list.
+	void MoveFlagshipBegin(vector<shared_ptr<Ship>> &ships, const shared_ptr<Ship> &flagship)
+	{
+		if(flagship)
+		{
+			auto it = find(ships.begin(), ships.end(), flagship);
+			if(it != ships.begin() && it != ships.end())
+			{
+				ships.erase(it);
+				ships.insert(ships.begin(), flagship);
```

Comment:
This is a bit inefficient since it involves moving every element in the vector. You can use [`std::move_backward`](https://en.cppreference.com/w/cpp/algorithm/move_backward) to only move every ship (saving + setting the first element to the saved flagship after the move).